---
title: "MHV Update"
author: "Evan Lester"
date: "July 5, 2016"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Overview
### Review of Experiments
* 2'3' Cyclic Phosphate RNA seq carried out by Susan Weiss' lab at Penn where Mouse BMMs (WT RNaseL and RNaseL -/-) were infected for either 9 or 12 hours with either WT MHV or NS2 mutant MHV
    + NS2 is a protein encoded by MHV that has 2’-5’-phosphodiesterase activity and has been shown to cleave 2'-5'-Oligoadenylate, a molecule that activates RNaseL. (NS2 is able to "completely antagonize RNaseL," and is required for MHV spread and hepatitis induction invivo [^1])
    + Daphne Cooper has done some preliminary analysis of this data, but nothing was published and it doesn't seem like she generated any findings
* 2'3' Cyclic Phosphate RNA seq mock infection that can be used as a negative control carried out by Dan Stetson's lab at the University of Washington

####Background
* Published cleavage sites
    + RNaseL: 
        - UU and UA dinucleotide cleavage [^2]
        - Cleavage at UU541 and UU743 in the mouse 18S ribosome [^3]
    + NendoU: 
        - hydrolyzes ssRNA and dsRNA substrates, preferably 3’ of uridylates and cleaves similar as XendoU and RNaseA, as the reaction products possess 2,3 cyclic phosphate ends.[^4] 
        - may digest RNA-PAMPs to avoid activation of the innate immune system -> possible role as a IFN-antagonist.[^5]
        - “it remains elusive if the NendoU substrate is RNA of viral or cellular origin and under which circumstances the utilization of an EndoU is required, as the enzyme is absent in invertebrate Nidoviruses."[^6]
* MHV genome

[^1]: Zhao, L. et al. Antagonism of the Interferon-Induced OAS-RNase L Pathway by Murine Coronavirus ns2 Protein Is Required for Virus Replication and Liver Pathology. Cell Host & Microbe 11, 607–616 (2012). 
[^2]: Cooper, D. A., Jha, B. K., Silverman, R. H., Hesselberth, J. R. & Barton, D. J. Ribonuclease L and metal-ion-independent endoribonuclease cleavage sites in host and viral RNAs. Nucleic Acids Research 42, 5202–5216 (2014).
[^3]: Cooper, D. A., Jha, B. K., Silverman, R. H., Hesselberth, J. R. & Barton, D. J. Ribonuclease L and metal-ion-independent endoribonuclease cleavage sites in host and viral RNAs. Nucleic Acids Research 42, 5202–5216 (2014).
[^4]: Kindler, E. & Thiel, V. ScienceDirect To sense or not to sense viral RNA — essentials of coronavirus innate immune evasion. Current Opinion in Microbiology 20, 69–75 (2014).
[^5]: Ibid.
[^6]: Ibid.

###Objectives
####Long Term

1. RNaseL: Identify regions of the MHV genome that are cleaved by RNaseL.
2. NendoU: Indentify cleavage patterns and substrates for NendoU cleavage in both the MHV and mouse genomes.
3. Angiogenin: Determine whether angiogenin (or another endonuclease) cleaves all tRNAs at the same rate following MHV infection or if the process is biased towards specific tRNAs.

####Short Term

* Ensure my cleavage location and dinucleotide plots align with Daphne's 
* Ensure that we can find evidence published RNaseL cleavage sites in the data
    + UU and UA dinucleotide cleavage
    + Cleavage at UU541 and UU743 in the mouse 18S ribosome
* Find regions of MHV that are cleaved by RNaseL
    + look for evidence of this in subtractive samples s5s7 and s6s8 and align these cleavage sites to the mhv genome to see what they are targeting
* 
    
### Current Progress
* Replicated Daphne's figures and made some alterations
    + Steps in analysis
        1. trim off adapters from fastq file and put UMI in headder
        2. create bowtie index of fasta reference file using bowtie-build
        3. align and sort fastq with trimmed seq to bowtie index files using bowtie2
        4. break up aligned bam files into + and - strands using 2nd column in bam file 16 = positive strand, 0 = negative strand
        5. dinucleotide mapping
        6. calculate dinucleotide frequency in FastA reference file
        7. multiply samples by dibuleotide frequency
        8. normalize read values to the total number of UMI reads 
        9. create subtraction pairs
    + Normalized each sample to the total number of UMI reads. (She normalized to a hard coded number that I haven't been able to identify)
    + Multiplied each dinucleotide pair by it's frequency in the FastA reference genome
    + 
    
    
###Figures

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
